dataland-framework-toolbox/src/main/kotlin/org/dataland/frameworktoolbox/frameworks/eutaxonomynonfinancials/custom/EuTaxonomyAssuranceComponent.kt

```diff
@@ -0,0 +1,82 @@
+package org.dataland.frameworktoolbox.frameworks.eutaxonomynonfinancials.custom
+
+import org.apache.commons.text.StringEscapeUtils
+import org.dataland.frameworktoolbox.intermediate.FieldNodeParent
+import org.dataland.frameworktoolbox.intermediate.components.ComponentBase
+import org.dataland.frameworktoolbox.intermediate.components.addStandardCellWithValueGetterFactory
+import org.dataland.frameworktoolbox.intermediate.components.addStandardUploadConfigCell
+import org.dataland.frameworktoolbox.intermediate.components.requireDocumentSupportIn
+import org.dataland.frameworktoolbox.intermediate.datapoints.NoDocumentSupport
+import org.dataland.frameworktoolbox.specific.fixturegenerator.elements.FixtureSectionBuilder
+import org.dataland.frameworktoolbox.specific.uploadconfig.elements.UploadCategoryBuilder
+import org.dataland.frameworktoolbox.specific.viewconfig.elements.SectionConfigBuilder
+import org.dataland.frameworktoolbox.specific.viewconfig.elements.getTypescriptFieldAccessor
+import org.dataland.frameworktoolbox.specific.viewconfig.functional.FrameworkDisplayValueLambda
+import org.dataland.frameworktoolbox.utils.typescript.TypeScriptImport
+
+/**
+ * Represents the EuTaxonomy-Specific "Assurance" component
+ */
+class EuTaxonomyAssuranceComponent(
+    identifier: String,
+    parent: FieldNodeParent,
+) : ComponentBase(
+    identifier, parent,
+    "org.dataland.datalandbackend.frameworks" +
+        ".eutaxonomynonfinancials.custom.AssuranceDataPoint",
+) {
+
+    override fun generateDefaultViewConfig(sectionConfigBuilder: SectionConfigBuilder) {
+        requireDocumentSupportIn(setOf(NoDocumentSupport))
+        sectionConfigBuilder.addStandardCellWithValueGetterFactory(
+            this,
+            FrameworkDisplayValueLambda(
+                "formatAssuranceForDataTable(${getTypescriptFieldAccessor(true)}, \"${
+                    StringEscapeUtils.escapeEcmaScript(
+                        label,
+                    )
+                }\")",
+                setOf(
+                    TypeScriptImport(
+                        "formatAssuranceForDataTable",
+                        "@/components/resources/dataTable/conversion/EutaxonomyAssuranceValueGetterFactory",
+                    ),
+                ),
+            ),
+        )
+        createNewViewConfigCellForAssuranceProvider(sectionConfigBuilder)
+    }
+
+    override fun generateDefaultUploadConfig(uploadCategoryBuilder: UploadCategoryBuilder) {
```

Comment:
Why does this have an upload config?

Reply:
Theoretisch unnötig, weil die uploadConfig ja statisch existiert und nicht autogeneriert wird. Richtig.

Aber:  Wir haben sowieso ein FormField für dieses Feld, also dachten wir uns dass wir hier einfach schonmal die uploadConfigGeneration korrekt hinterlegen, weil man dann irgendwann in der Zukunft (wenn man wieder hierherkommt) nicht erneut suchen muss, wie das FormField hieß.